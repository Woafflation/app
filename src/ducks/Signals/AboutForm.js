import React from 'react'
import PropTypes from 'prop-types'
import { Formik, Form } from 'formik'
import { Button } from '@santiment-network/ui'
import FormikInput from './../../components/formik-santiment-ui/FormikInput'
import styles from './TriggerForm.module.scss'

const AboutForm = ({
  title = `Signal_[${new Date().toLocaleDateString('en-US')}]`,
  description = '',
  isEdit = false,
  onSubmit,
  onBack
}) => (
  <Formik
    initialValues={{ title, description }}
    isInitialValid
    validate={values => {
      let errors = {}
      if (!values.title) {
        errors.title = 'Required'
      } else if (values.title.length < 3) {
        errors.title = 'Title has to be longer than 2 characters'
      }
      return errors
    }}
    onSubmit={values => onSubmit(values)}
  >
    {({ isSubmitting, isValid }) => (
      <Form>
        <div className={styles.Field}>
          <label>Name of the signal</label>
          <FormikInput name='title' type='text' placeholder='Name of signal' />
        </div>
        <div className={styles.Field}>
          <label>Description</label>
          <FormikInput
            name='description'
            type='text'
            placeholder='Description of the signal'
          />
        </div>
        <div className={styles.controls}>
          <Button
            type='button'
            variant={'flat'}
            accent='normal'
            border
            onClick={onBack}
          >
            Back
          </Button>
          <Button
            type='submit'
            disabled={!isValid || isSubmitting}
            isActive={isValid && !isSubmitting}
            variant={'fill'}
            accent='positive'
          >
            {isEdit ? 'Update' : 'Create'}
          </Button>
        </div>
      </Form>
    )}
  </Formik>
)

AboutForm.propTypes = {
  onSubmit: PropTypes.func.isRequired
}

export default AboutForm
