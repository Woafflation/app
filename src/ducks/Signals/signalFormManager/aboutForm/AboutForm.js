import React, { useState } from 'react'
import PropTypes from 'prop-types'

import { Formik, Form } from 'formik'
import Label from '@santiment-network/ui/Label'
import Button from '@santiment-network/ui/Button'
import FormikInput from '../../../../components/formik-santiment-ui/FormikInput'
import styles from '../signalCrudForm/signal/TriggerForm.module.scss'

const AboutForm = ({ triggerMeta, isEdit = false, onSubmit, onBack }) => {
  const [trigger, setTrigger] = useState(triggerMeta)
  return (
    <Formik
      initialValues={trigger}
      isInitialValid
      validate={values => {
        let errors = {}
        if (!values.title) {
          errors.title = 'Required'
        } else if (values.title.length < 3) {
          errors.title = 'Title has to be longer than 2 characters'
        } else if (values.title.length > 120) {
          errors.title = 'Title has to be less than 120 characters'
        }
        if (values.description.length > 240) {
          errors.description = 'Description has to be less than 240 characters'
        }
        return errors
      }}
      onSubmit={values => {
        onSubmit(values)
      }}
    >
      {({ values: { description = '' }, isSubmitting, isValid }) => (
        <Form className={styles.AboutForm}>
          <div className={styles.triggerFormItem}>
            <div className={styles.row}>
              <div className={styles.Field}>
                <Label accent='waterloo' className={styles.label}>
                  Name of the signal
                </Label>
                <FormikInput
                  name='title'
                  type='text'
                  placeholder='Name of the signal'
                  onChange={value =>
                    setTrigger({
                      ...trigger,
                      title: value
                    })
                  }
                />
              </div>
            </div>

            <div className={styles.row}>
              <div className={styles.Field}>
                <Label accent='waterloo' className={styles.label}>
                  Description ({description.length}/140)
                </Label>
                <FormikInput
                  name='description'
                  type='text'
                  maxLength={140}
                  placeholder='Description of the signal'
                  onChange={value => {
                    setTrigger({
                      ...trigger,
                      description: value
                    })
                  }}
                />
              </div>
            </div>
          </div>

          <div className={styles.controls}>
            <Button
              type='button'
              variant={'flat'}
              accent='grey'
              border
              onClick={() => {
                onBack(trigger)
              }}
            >
              Back
            </Button>
            <Button
              type='submit'
              disabled={!isValid || isSubmitting}
              isActive={isValid && !isSubmitting}
              variant={'fill'}
              accent='positive'
            >
              Confirm
            </Button>
          </div>
        </Form>
      )}
    </Formik>
  )
}

AboutForm.propTypes = {
  onSubmit: PropTypes.func.isRequired
}

export default AboutForm
