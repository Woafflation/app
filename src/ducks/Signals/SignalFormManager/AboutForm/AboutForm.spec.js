/* eslint-env jest */
import React from 'react'
import { shallow, mount } from 'enzyme'
import toJson from 'enzyme-to-json'
import AboutForm from './AboutForm'

describe('AboutForm', () => {
  xit('render correctly text component', () => {
    const constantDate = new Date('2017-06-13T04:41:20')

    /* eslint no-global-assign:off */
    Date = class extends Date {
      constructor () {
        return constantDate
      }
    }
    const wrapper = shallow(
      <InfoSignalForm onBack={() => {}} onSubmit={() => {}} />
    )
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
