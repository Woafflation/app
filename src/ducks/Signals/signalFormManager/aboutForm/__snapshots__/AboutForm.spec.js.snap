// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AboutForm render correctly text component 1`] = `
<Formik
  enableReinitialize={true}
  initialValues={
    Object {
      "description": "Custom Description",
      "title": "Title of Trigger",
    }
  }
  isInitialValid={true}
  onSubmit={[Function]}
  validate={[Function]}
  validateOnBlur={true}
  validateOnChange={true}
>
  <Component />
</Formik>
`;
